spacy/pipeline/trainable_pipe.pxd

```diff
@@ -0,0 +1,8 @@
+from .pipe cimport Pipe
+from ..vocab cimport Vocab
+
+cdef class TrainablePipe(Pipe):
+    cdef public Vocab vocab
```

Comment:
```suggestion
    cdef readonly Vocab vocab
```

Again it's tough to decide between public and readonly, maybe it's better to be flexible and allow public. But segfaults suck, and we don't want to worry that this is a problem, so let's just do readonly.

Reply:
It looks like we need to keep this one `public` - the factories (parsing `nlp` from config) seem to not work otherwise.